### PR TITLE
fix(h2): improve I/O errors emitted by H2Upgraded

### PR DESCRIPTION
**Pull request by nox**

### CODE REVIEW

File: src/proto/h2/mod.rs [hunk 1]
```diff
@@ -335,21 +339,34 @@ where
...
-            Some(Ok(cnt)) => self.send_stream.write(&buf[..cnt], false).map(|()| cnt),
-            Some(Err(e)) => Err(h2_to_io_error(e)),
-        })
+
+        let cnt = match ready!(self.send_stream.poll_capacity(cx)) {
+            None => Some(0),
+            Some(Ok(cnt)) => self
+                .send_stream
+                .write(&buf[..cnt], false)
+                .ok()
```
Comment (seanmonstar) [inline src/proto/h2/mod.rs:352]: This will ignore the error, but that's on purpose, right?

**Review by seanmonstar — APPROVED** [review]
I think this looks right.

(Have you been able to try out this in your app to see it improves it for you?)

File: src/proto/h2/mod.rs (hunk 1 above)
Comment (nox) [reply src/proto/h2/mod.rs:352]: Yeah, that's the intent, as we will get the right error from `poll_reset` anyway.

Comment (seanmonstar) [reply src/proto/h2/mod.rs:352]: Ok got it. Perhaps a comment to that effect will help us when looking in this file a few months or years later 😅

Comment (nox) [reply src/proto/h2/mod.rs:354]: Done.

**Pull request merged**